nextjs-app/app/components/hero/HeroBlock.tsx

```diff
@@ -0,0 +1,148 @@
+            variant: header.primaryButton.variant as ButtonVariant,
+          }
+        : undefined,
+      secondaryButton: header.secondaryButton
+        ? {
+            label: header.secondaryButton.label,
+            url: '#', // TODO: Replace with resolveUrl(header.secondaryButton) when WEB-4279 is implemented
+            icon: header.secondaryButton.icon,
+            variant: header.secondaryButton.variant as ButtonVariant,
+          }
+        : undefined,
+    }),
+    [header],
+  )
+
+  const getLayoutClasses = () => {
+    switch (layoutEnum) {
+      case HeroLayout.TWO_COLUMN_IMAGE_LEFT:
+      case HeroLayout.TWO_COLUMN_IMAGE_RIGHT:
+        return `flex flex-col ${imageContained ? 'gap-12' : 'gap-0'} lg:grid lg:grid-cols-2 lg:gap-20 lg:items-center`
+      case HeroLayout.IMAGE_FULL_WIDTH:
+      case HeroLayout.IMAGE_CONTAINED:
+        return 'relative'
+      default:
+        return ''
+    }
+  }
+
+  const getOrderClasses = (type: 'header' | 'image') => {
+    if (layoutEnum !== HeroLayout.TWO_COLUMN_IMAGE_RIGHT) return ''
+    return type === 'header' ? 'lg:order-2' : 'lg:order-1'
+  }
+
+  const isOverlay =
+    layoutEnum === HeroLayout.IMAGE_FULL_WIDTH ||
+    layoutEnum === HeroLayout.IMAGE_CONTAINED
+
+  const isTwoColumn =
+    layoutEnum === HeroLayout.TWO_COLUMN_IMAGE_LEFT ||
+    layoutEnum === HeroLayout.TWO_COLUMN_IMAGE_RIGHT
+
+  const standardPadding = isTwoColumn && !imageContained ? 'py-0' : 'py-20'
+
+  const horizontalPaddingClasses =
+    isTwoColumn && !imageContained
+      ? 'px-0'
+      : isTwoColumn && imageContained
+        ? 'px-5 sm:px-10 xl:px-20 2xl:px-20'
+        : 'px-5 sm:px-10 xl:px-20 2xl:px-[clamp(80px,4vw,160px)]'
+
+  const headerMargin = isTwoColumn
+    ? imageContained
+      ? 'mb-5 sm:mb-10'
+      : 'my-5 sm:my-10'
+    : ''
+
+  return (
+    <section style={{ backgroundColor }}>
+      {isOverlay && hasImage ? (
+        <div className="relative min-h-[500px]">
+          {layoutEnum === HeroLayout.IMAGE_FULL_WIDTH ? (
+            <div className="absolute inset-0">
+              <HeroImage image={image} layout={layoutEnum} />
+            </div>
+          ) : (
+            <div className="absolute inset-0 flex justify-center px-5 sm:px-10 md:px-20">
+              <div className="w-full max-w-[1376px] py-5 sm:py-10 md:py-20">
+                <HeroImage image={image} layout={layoutEnum} />
+              </div>
+            </div>
+          )}
+          <div className="relative z-10 flex items-center justify-center min-h-[500px] px-10 md:px-20 py-20">
+            <div
+              className={`w-full max-w-4xl ${layoutEnum === HeroLayout.IMAGE_CONTAINED ? 'max-[1100px]:px-5' : ''}`}
+            >
+              <BlockHeaderSection
+                header={transformedHeader}
+                backgroundColor={BackgroundTheme.DARK}
+                headerAlignment={alignmentEnum}
+              />
+            </div>
+          </div>
+        </div>
```

Comment:
Prolly good idea to break this out to a `HeroBlockWithImage` component. The cyclical complexity here is gettin' pretty high already, and will likely just increase in the future.